central/telemetry/centralclient/instance_config.go

```diff
@@ -33,9 +38,60 @@ var (
 	enabled  bool
 )
 
+const (
+	disabledKey = "DISABLED"
+)
+
+type remoteConfig struct {
+	Key string `json:"storage_key_v1,omitempty"`
+}
+
+func downloadConfig(u string) (*remoteConfig, error) {
+	req, err := http.NewRequest(http.MethodGet, u, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot request telemetry configuration")
+	}
+	client := http.Client{
+		Timeout:   5 * time.Second,
+		Transport: proxy.RoundTripper(),
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot download telemetry configuration")
+	}
+	var cfg *remoteConfig
+	err = json.NewDecoder(resp.Body).Decode(&cfg)
+	return cfg, errors.Wrap(err, "cannot decode telemetry configuration")
+}
+
+func isReleaseBuild() bool {
+	return buildinfo.ReleaseBuild &&
+		version.GetMainVersion() != "" &&
+		!strings.Contains(version.GetMainVersion(), "-")
+}
+
 func getInstanceConfig() (*phonehome.Config, map[string]any, error) {
 	key := env.TelemetryStorageKey.Setting()
-	if key == "" || env.OfflineModeEnv.BooleanSetting() {
+	if key == disabledKey || env.OfflineModeEnv.BooleanSetting() {
+		return nil, nil, nil
+	}
+
+	if env.TelemetryConfigURL.Setting() != "" &&
```

Comment:
how about splitting condition. I understand we can return if `env.TelemetryConfigURL.Setting() == ""`